### DISCUSSION TITLE
Configure custom LSP using filetype plugin

### DISCUSSION THREAD

**Discussion by sagikazarmark** [LunarVim/LunarVim#1956] [General]
I'm trying to configure `rnix` LSP server (supported by [nvim-lspconfig](https://github.com/neovim/nvim-lspconfig/blob/master/lua/lspconfig/rnix.lua)).

I installed the `rnix-lsp` binary manually (it's not supported by [nvim-lsp-installer](https://github.com/williamboman/nvim-lsp-installer) and added the following to `config.lua`:

```lua
require("lvim.lsp.manager").setup("rnix");
```

Unfortunately that didn't work, so [based on the documentation](https://www.lunarvim.org/languages/#overriding-the-default-setup-options) I added this:

```lua
require("lspconfig")["rnix"].setup({})
```

This seemed to work.

The next thing I tried is adding the the configuration to filetype plugin [based on the configuration](https://www.lunarvim.org/configuration/07-ftplugin.html) at `$LUNARVIM_CONFIG_DIR/after/ftplugin/nix.lua` (which I translated to `~/.config/lvim/after/ftplugin/nix.lua` [based on the code](https://github.com/LunarVim/LunarVim/search?q=LUNARVIM_CONFIG_DIR)):

```lua
require("lspconfig")["rnix"].setup({})
```

That seemed to work fine.

The reason why I wanted to write this down is because it wasn't immediately obvious how to do this (eg `require("lvim.lsp.manager")` vs `require("lspconfig")` or how to resolve `$LUNARVIM_CONFIG_DIR`), so hopefully this will help others and it's searchable on GitHub.

Ultimately, this should probably go into the documentation. Most of the information is available in different places, a single guide might be better.

An interesting piece of information: there is a plan to [Support ALL THE LSP](https://github.com/LunarVim/LunarVim/discussions/1784).

I'm not completely sure if this is the right way to do it, so if you spot something wrong, please let me know.

It's also unclear whether using `require("lspconfig")` directly is a good idea (although it's referenced in the documentation) or why `require("lvim.lsp.manager")` wouldn't work (although looking at the code, it seems some kind of "provider" is missing?).

**Comment by kylo252** [comment]
This is now fixed in #1981

**Reply by sagikazarmark** [reply]
Awesome, thanks!
